drivers/serial/uart_mcux_flexcomm.c

```diff
@@ -262,59 +262,59 @@ static int usart_mcux_lpc_init(struct device *dev)
 	return 0;
 }
 
-static const struct uart_driver_api usart_mcux_lpc_driver_api = {
-	.poll_in = usart_mcux_lpc_poll_in,
-	.poll_out = usart_mcux_lpc_poll_out,
-	.err_check = usart_mcux_lpc_err_check,
+static const struct uart_driver_api mcux_flexcomm_driver_api = {
+	.poll_in = mcux_flexcomm_poll_in,
+	.poll_out = mcux_flexcomm_poll_out,
+	.err_check = mcux_flexcomm_err_check,
 #ifdef CONFIG_UART_INTERRUPT_DRIVEN
-	.fifo_fill = usart_mcux_lpc_fifo_fill,
-	.fifo_read = usart_mcux_lpc_fifo_read,
-	.irq_tx_enable = usart_mcux_lpc_irq_tx_enable,
-	.irq_tx_disable = usart_mcux_lpc_irq_tx_disable,
-	.irq_tx_complete = usart_mcux_lpc_irq_tx_complete,
-	.irq_tx_ready = usart_mcux_lpc_irq_tx_ready,
-	.irq_rx_enable = usart_mcux_lpc_irq_rx_enable,
-	.irq_rx_disable = usart_mcux_lpc_irq_rx_disable,
-	.irq_rx_ready = usart_mcux_lpc_irq_rx_ready,
-	.irq_err_enable = usart_mcux_lpc_irq_err_enable,
-	.irq_err_disable = usart_mcux_lpc_irq_err_disable,
-	.irq_is_pending = usart_mcux_lpc_irq_is_pending,
-	.irq_update = usart_mcux_lpc_irq_update,
-	.irq_callback_set = usart_mcux_lpc_irq_callback_set,
+	.fifo_fill = mcux_flexcomm_fifo_fill,
+	.fifo_read = mcux_flexcomm_fifo_read,
+	.irq_tx_enable = mcux_flexcomm_irq_tx_enable,
+	.irq_tx_disable = mcux_flexcomm_irq_tx_disable,
+	.irq_tx_complete = mcux_flexcomm_irq_tx_complete,
+	.irq_tx_ready = mcux_flexcomm_irq_tx_ready,
+	.irq_rx_enable = mcux_flexcomm_irq_rx_enable,
+	.irq_rx_disable = mcux_flexcomm_irq_rx_disable,
+	.irq_rx_ready = mcux_flexcomm_irq_rx_ready,
+	.irq_err_enable = mcux_flexcomm_irq_err_enable,
+	.irq_err_disable = mcux_flexcomm_irq_err_disable,
+	.irq_is_pending = mcux_flexcomm_irq_is_pending,
+	.irq_update = mcux_flexcomm_irq_update,
+	.irq_callback_set = mcux_flexcomm_irq_callback_set,
 #endif
 };
 
-#ifdef CONFIG_USART_MCUX_LPC_0
+#ifdef CONFIG_UART_MCUX_FLEXCOMM_0
 #ifdef CONFIG_UART_INTERRUPT_DRIVEN
-static void usart_mcux_lpc_config_func_0(struct device *dev);
+static void mcux_flexcomm_config_func_0(struct device *dev);
 #endif
 
-static const struct usart_mcux_lpc_config usart_mcux_lpc_0_config = {
+static const struct mcux_flexcomm_config mcux_flexcomm_0_config = {
 	.base = (USART_Type *)DT_USART_MCUX_LPC_0_BASE_ADDRESS,
 	.clock_source = 0,
 	.baud_rate = DT_USART_MCUX_LPC_0_BAUD_RATE,
```

Comment:
I would replace these lines with the following:
```
	.base = (USART_Type *)DT_NXP_LPC_USART_USART_0_BASE_ADDRESS,
	.clock_source = kCLOCK_FlexComm0,
	.baud_rate = DT_NXP_LPC_USART_USART_0_CURRENT_SPEED,
```
Or their equivalent Flexcomm name and drop USART defines in fixup.

Reply:
Done